node-it/src/test/scala/com/wavesplatform/it/sync/smartcontract/InvokeScriptTransactionSuite.scala

```diff
@@ -110,6 +156,36 @@ class InvokeScriptTransactionSuite extends BaseTransactionSuite with CancelAfter
     }
   }
 
+  test("contract caller invokes a function on a contract by alias") {
+    val arg               = ByteStr(Array(43: Byte))
+
+    val _ = sender.invokeScript(
+      caller,
+      "alias:I:alias",
+      func = Some("foo"),
+      args = List(CONST_BYTESTR(arg).explicitGet()),
+      payment = Seq(),
+      fee = 1.waves,
+      waitForTx = true
+    )
+
+    sender.getDataByKey(firstContract, "a") shouldBe BinaryDataEntry("a", arg)
+    sender.getDataByKey(firstContract, "sender") shouldBe BinaryDataEntry("sender", ByteStr.decodeBase58(caller).get)
+  }
+
+  test("enable use this with address") {
+    sender.invokeScript(
+      caller,
+      firstContract,
+      func = Some("baz"),
+      args = List(),
+      payment = Seq(),
+      fee = 1.waves,
+      waitForTx = true
+     )
+    sender.getDataByKey(firstContract, "test") shouldBe BinaryDataEntry("test", ByteStr.decodeBase58(firstContract).get)
+  }
```

Comment:
Add test where dApp is called by alias and call "baz" function, which uses "this" expression.